assets/js/src/utils/index.js

```diff
@@ -26,6 +26,23 @@ export const getProductFieldObject = ( product, quantity ) => {
 		),
 		...variantData,
 	};
+
+	// Apply discounts to ecommerce events.
+	// https://developers.google.com/analytics/devguides/collection/ga4/apply-discount?client_type=gtag
+	if (
+		typeof product.price_after_coupon_discount === 'number' &&
+		! isNaN( product.price_after_coupon_discount ) &&
```

Comment:
Why do we need these two? `product.price_after_coupon_discount` seems to be always a number

Reply:
Since there were not only purchase event would call the method `getProductFieldObject()`, for example `begin_checkout` event would call this method as well. Initially I wanted to test if `product.price_after_coupon_discount` is defined but since the discount could be `0` and `0` is treated as `false`, that's why I added these two conditions here.

Reply:
@puntope 

> But what about the NaN line check?

I added it because I see `typeof NaN` is also a `number`, so added it for some prevention.

> Also WDYT of doing something like:
if ( product?.price_after_coupon_discount < product.prices.price  ) 
So we can save both checks number and nan
You can test it but in case product.price_after_coupon_discount is undefined this would be false still.

Ah good point, thanks for the info.